train/test.cs

```diff
@@ -0,0 +1,26 @@
+﻿using NUnit.Framework;
+using System.IO;
+using System;
+
+namespace MED_test
+{
+    public class Tests
+    {
+        private int[] Expected = new int[] { 5, 1, 4, 4 };
+        private string[] str1 = new string[] { "intention", "abc", "hello", "lion" };
+        private string[] str2 = new string[] { "execution", "dbc", "hi", "emotion" };
+        [SetUp]
+        public void Setup()
+        {
+        }
+        [Test]
+        public void TestMethod1()
+        {
```

Comment:
It's good that you set different set of values to test your MED method. Besides, I think you need extra test cases about values out of the valid bounds, e.g.
assert MED(null, a valid string) / MED(null, null) VS. -1

Furthermore, if there is any exception that could be thrown by your method when invalid inputs are given, you can also catch a specific exception in your unit test, and then assert if its type is the exception type your expected.